h2/utilities.py

```diff
@@ -67,6 +67,20 @@ def secure_headers(headers):
             yield header
 
 
+def extract_method_header(headers):
+
+    request_header = (
+            'GET', 'HEAD',
+            'POST', 'PUT'
+            'DELETE', 'TRACE',
+            'CONNECT',
+            )
+
+    for k, v in headers:
+        if k == ':method' and v in request_header:
+            return (k, v)
```

Comment:
So we can change this function up a bit.

Firstly, let's not be too fussy about what `:method` we get. If we get a `:method`, we use it, even if it's not something we recognise.

Secondly, let's not return the tuple: just the method name.

Thirdly, this doesn't handle the possibility of receiving unicode and bytes headers. You'll want to look at some of the other functions in this module to see what they do: see the use of `_SECURE_HEADERS`, for example.